src/bin/main.rs

```diff
@@ -464,6 +472,14 @@ fn determine_operation(matches: &Matches) -> Result<Operation, OperationError> {
         if minimal_config_path.is_some() {
             return Err(OperationError::MinimalPathWithStdin);
         }
+        if matches.opt_present("check") {
+            return Err(OperationError::CheckWithStdin);
+        }
+        if let Some(mode) = matches.opt_str("emit") {
+            if mode == "json" {
```

Comment:
All of the emit modes other than `stdout` will currently fail with stdin (not just json), so I think this should check for any emit mode other than stdout and fail accordingly

Reply:
Thanks, I'll tweak it.

Reply:
Done.

Reply:
This is a bit broken, since the default is `--emit=files`.  `format_string` just overrides whatever it was with `stdout`, but I think it needs to distinguish between an explicit `--emit=files` (which would be an error) and not being specified (in which case default to `stdout`), which suggests that `options.emit_mode` should be an `Option`.

Reply:
(Fixed again)